src/ArmTemplates/Extractor/EntityExtractors/ParametersExtractor.cs

```diff
@@ -119,6 +120,39 @@ async Task AddServiceUrlParameterAsync()
                 parameters.Add(ParameterNames.ServiceUrl, new TemplateObjectParameterProperties() { Value = serviceUrls });
             }
 
+            async Task AddApiOauth2ScopeParameterAsync()
+            {
+                if (!extractorParameters.ParametrizeApiOauth2Scope)
+                {
+                    return;
+                }
+
+                var apiOauth2Scopes = new Dictionary<string, string>();
+                foreach (var apiName in apisToExtract)
+                {
+                    var apiDetails = await this.apisClient.GetSingleAsync(apiName, extractorParameters);
+
+                    if (apiDetails.Properties.AuthenticationSettings?.OAuth2 is not null)
+                    {
+                        string apiOAuthScope;
+                        var validApiName = ParameterNamingHelper.GenerateValidParameterName(apiName, ParameterPrefix.Api);
+
+                        if (extractorParameters.ApiOauth2ScopeParameters is null)
+                        {
+                            apiOAuthScope = apiDetails.Properties.AuthenticationSettings.OAuth2?.Scope;
+                        }
+                        else
+                        {
+                            apiOAuthScope = extractorParameters.ApiOauth2ScopeParameters.FirstOrDefault(x => x.ApiName.Equals(apiName))?.Scope;
```

Comment:
`x.ApiName.Equals(apiName)` - can troubles appear here because of string comparison? Maybe use `String.OrdinalIgnoreCase` or some other overload?

Reply:
forgot to place link to doc - https://docs.microsoft.com/en-us/dotnet/api/system.stringcomparison?view=net-6.0